enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,350 @@
+will be populated by the installer using the entries from `platform.azure.userTags` field of `install-config`.
+
+`status.platformStatus.azure` of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create Azure resources will apply these tags to all Azure 
+resources they create. List of in-cluster operators managing cloud resources
+could vary across platform types, example for AWS there are additional operators
+like aws-ebs-csi-driver-operator, aws-efs-csi-driver-operator to manage specific
+resources.
+
+| Operator | Resources created by the operator |
+| -------- | ----------------------------- |
+| cluster-image-registry-operator | Storage Account |
+| cluster-ingress-operator | DNS records |
+| machine-api-provider-azure | Application Security Group, Availability Set, Group, Load Balancer, Public IP Address, Route, Network Security Group, Virtual Machine Extension, Virtual Interface, Virtual Machine, Virtual Network. |
+
+Below list of terraform Azure APIs to create resources should be updated to add user
+defined tags and as well the openshift default tag in the installer component.
+`azurerm_resource_group, azurerm_image, azurerm_lb, azurerm_network_security_group, azurerm_storage_account, azurerm_user_assigned_identity, azurerm_virtual_network, azurerm_linux_virtual_machine, azurerm_network_interface, azurerm_dns_cname_record`
+
+API update example:
+A local variable should be defined, which merges the default tag and the user
+defined Azure tags, which should be referenced in the Azure resource APIs.
+``` terraform
+locals {
+  tags = merge(
+    {
+      "kubernetes.io_cluster.${var.cluster_id}" = "owned"
+    },
+    var.azure_extra_tags,
+  )
+}
+
+resource "azurerm_resource_group" "main" {
+  tags     = local.tags
+}
+```
+
+The tag of the form `kubernetes.io_cluster.<cluster_id>:owned` added by openshift, where
+cluster_id is a string generated by concatenating user inputted cluster name and a random 
+string will be limited to a maximum length of 27 characters by trimming long cluster name 
+to 21 characters.
+
+#### Caveats
+1. Updating or removing resource tags added by Openshift, using an external interface
+   may or may not be reconciled by the operator managing the resource.
+2. Updating tags of individual resources is not supported and any tag present in 
+   `.status.platformStatus.azure.resourceTags` of `infrastructure.config.openshift.io/v1` resource 
+   will result in adding tags to all Openshift managed Azure resources. 
+
+### Risks and Mitigations
+
+### Drawbacks
+- User-defined tags cannot be updated on an Azure resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in the scope.
+  User-defined tags can be updated on the following Azure resources.
+  - ResourceGroup
+  - Storage Account
+  - DNS Zones
+  - DNS Records
```

Comment:
I went through [Microsoft's docs](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support) to double-check, and it wasn't clear to me that "DNS records" had tag support (e.g. it says that dnszones/A records do not, privateDnsZones /A records do not, etc.). Can you confirm that tags are supported?

Reply:
Yes, these resources do not have explicit tag field, but it's added as metadata for these resources.